app/src/main/java/com/work/restaurant/util/DateAndTime.kt

```diff
@@ -0,0 +1,165 @@
+    }
+
+
+    fun beforeWeek(date: List<String>): String {
+
+        val calendar = Calendar.getInstance()
+        calendar.set(date[0].toInt(), date[1].toInt() - 1, date[2].toInt())
+        calendar.add(Calendar.DATE, -7)
+
+        return simpleDateFormat.format(calendar.time)
+    }
+
+    fun afterDate(date: List<String>): String {
+
+        val calendar = Calendar.getInstance()
+        calendar.set(date[0].toInt(), date[1].toInt() - 1, date[2].toInt())
+        calendar.add(Calendar.DATE, 1)
+
+        return simpleDateFormat.format(calendar.time)
+    }
+
+
+    fun afterWeek(date: List<String>): String {
+        val calendar = Calendar.getInstance()
+        calendar.set(date[0].toInt(), date[1].toInt() - 1, date[2].toInt())
+        calendar.add(Calendar.DATE, 7)
+
+        return simpleDateFormat.format(calendar.time)
+    }
+
+
+    fun currentTime(): String {
+        val currentTime = Calendar.getInstance().time
+        return SimpleDateFormat("ahhmm", Locale.getDefault()).format(currentTime)
+    }
+
+    fun lastDayOfThisMonth(): Int =
+        Calendar.getInstance().getMaximum(Calendar.DAY_OF_MONTH)
+
+
+    fun convertSaveTime(time: String): String {
+
+        val splitTime = time.split(" ")
+
+        val splitHour = if (splitTime[1].substring(
+                0,
+                splitTime[1].length - 1
+            ).toInt() / 10 == 0
+        ) {
+            if (splitTime[0] == "오후") {
+                "${(splitTime[1].substring(0, splitTime[1].length - 1)).toInt() + 12}"
+            } else {
+                "0" + "${(splitTime[1].substring(0, splitTime[1].length - 1)).toInt()}"
+            }
+        } else {
+            if (splitTime[0] == "오후") {
+                if ((splitTime[1].substring(0, splitTime[1].length - 1)) == "12") {
+                    "${(splitTime[1].substring(0, splitTime[1].length - 1)).toInt()}"
+                } else {
+                    "${(splitTime[1].substring(0, splitTime[1].length - 1)).toInt() + 12}"
+                }
+            } else {
+                "${(splitTime[1].substring(0, splitTime[1].length - 1)).toInt()}"
+            }
+        }
+
+        val minute = splitTime[2].substring(
+            0,
+            splitTime[2].length - 1
+        )
+
+        return splitTime[0] + splitHour + minute
+    }
+
+    fun convertPickerTime(hour: Int, minute: Int): String {
+
+        val getAmPm = if (hour > 12) {
+            "오후"
+        } else {
+            "오전"
+        }
+        val getHour =
+            when (hour) {
+                0 -> {
+                    "${hour + 12}시"
+                }
+                in 1..12 -> {
+                    "${hour}시"
+                }
+                else -> {
+                    "${hour - 12}시"
+                }
+            }
+
+        val getMinute = if (minute / 10 == 0) {
+            "0${minute}분"
+        } else {
+            "${minute}분"
+        }
+        return "$getAmPm $getHour $getMinute"
```

Comment:
```suggestion
        val calendar = Calendar.getInstance().apply{
              set(Calendar.HOUR, hour)
              set(Calendar.MINUTE, minute)
        }
        return SimpleDateFormat("a h mm").format(calendar.date)
```
요런거 많이 찾아보세요. 제가 자주사용하는 방법이지만 더 좋은방법이 있을 수도?

Reply:
37305a0  수정하긴 했는데 오전 오후가 바뀌게 나오길래 조금 변경했습니다.